crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2789,10 +2789,35 @@ impl<'db> TypeInferenceBuilder<'db> {
         } = attribute;
 
         let value_ty = self.infer_expression(value);
-        value_ty
-            .member(self.db, &Name::new(&attr.id))
-            .ignore_possibly_unbound()
-            .unwrap_or(Type::Unknown)
+        match value_ty.member(self.db, &Name::new(&attr.id)) {
+            Symbol::Type(member_ty, boundness) => {
+                if boundness == Boundness::PossiblyUnbound {
+                    self.diagnostics.add(
+                        attribute.into(),
+                        "possibly-unbound-attribute",
+                        format_args!(
+                            "The attribute `{}` on type `{}` is possibly unbound",
```

Comment:
I'd go for a slightly shorter error message here:

```suggestion
                            "Attribute `{}` on type `{}` is possibly unbound",
```